src/test/java/ca/gc/cyber/ops/assemblyline/java/client/clients/AssemblylineClientTest.java

```diff
@@ -198,11 +196,10 @@ private void verifyExpectedPath(String expectedPath) {
 
             assertEquals("json", json.getFieldName());
             assertEquals("text/plain;charset=UTF-8", json.getContentType());
-            U jsonRequestPart = objectMapper.readValue(json.getString(), type);
-            assertEquals(expectedJsonRequest, jsonRequestPart);
+            JSONAssert.assertEquals(expectedJsonRequest, json.getString(), false);
```

Comment:
This is the primary change in the test class. It switches to comparing expected JSON with actual JSON instead of comparing an expected object (that represents how _we_ expect JSON to be deserialized) with actual object (that was deserialized from JSON).

There are two parts to the problem:
1. The old `equals()` methods for `SubmitMetadata` and `IngestBase` (genereated by Lombok) weren't looking at any fields, because the two subclasses of `SubmitBase` don't have any fields themselves and we didn't tell Lombok to use the superclass.
2. With the new `equals()`, tests were failing because our expected verison of the field `submitBase.metadata["key2"]` was an instance of `MetadataObjectTest`, but, beucause `metadata` is just `Map<String, Object>`, Jackson deserialized the _actual_ field as a `Map`, which was not equal to the expected instance of `MetadataObjectTest`.

Beyond the bug, this change removes an unnecessary layer of abstraction (the deserialization) from a check that is ultimately about JSON strings, not Java objects.